Core/include/Acts/Seeding/GNN_DataStorage.hpp

```diff
@@ -0,0 +1,275 @@
+    //   // }
+    //*dereferences pointer to then pass as ref 
+      m_etaBins.at(binIndex).m_vn.push_back(new TrigFTF_GNN_Node<space_point_t>(*sp.SP, min_tau, max_tau)); 
+    }
+    else {
+    //   // if (useML) {
+    //   //   const Trk::SpacePoint* osp = sp.offlineSpacePoint();
+    //   //   const InDet::PixelCluster* pCL = dynamic_cast<const InDet::PixelCluster*>(osp->clusterList().first);
+    //   //   float cluster_width = pCL->width().widthPhiRZ().y();
+    //   //   if(cluster_width > 0.2) return -3;
+    //   // }
+      m_etaBins.at(binIndex).m_vn.push_back(new TrigFTF_GNN_Node<space_point_t>(*sp.SP)); //add template here? add , min_tau, max_tau here for constructor 
+    }   
+
+    return 0;
+  }
+
+  //for safety to prevent passing as copy 
+  TrigFTF_GNN_DataStorage(const TrigFTF_GNN_DataStorage &) = delete;
+  TrigFTF_GNN_DataStorage &operator=(const TrigFTF_GNN_DataStorage &) = delete;
+
+  unsigned int numberOfNodes() const {
+
+    unsigned int n=0;
+    
+    for(auto& b : m_etaBins) {
+      n += b.m_vn.size();
+    }
+    return n;
+  }
+
+
+  void getConnectingNodes(std::vector<const TrigFTF_GNN_Node<space_point_t>*>& vn) {
+  
+    vn.clear();
+    vn.reserve(numberOfNodes());
+    
+    for(const auto& b : m_etaBins) {
+      for(typename std::vector<TrigFTF_GNN_Node<space_point_t>*>::const_iterator nIt = b.m_vn.begin();nIt!=b.m_vn.end();++nIt) {
+        if((*nIt)->m_in.empty()) continue;
+        if((*nIt)->m_out.empty()) continue;
+        vn.push_back(*nIt);
+      }
+    }
+  }
+
+
+  void sortByPhi() {
+    for(auto& b : m_etaBins) b.sortByPhi();
+  }
+
+  void generatePhiIndexing(float dphi) {
+    for(auto& b : m_etaBins) b.generatePhiIndexing(dphi);
+  }
+
+
+  const TrigFTF_GNN_EtaBin<space_point_t>& getEtaBin(int idx) const {
+    if(idx >= static_cast<int>(m_etaBins.size())) idx = idx-1;
+    return m_etaBins.at(idx);
+  }
+
+ protected:
+
+  const TrigFTF_GNN_Geometry<space_point_t>&  m_geo;
+
+  std::vector<TrigFTF_GNN_EtaBin<space_point_t>> m_etaBins; 
+
+};
+
+template <typename space_point_t>  
+class TrigFTF_GNN_Edge {
+ public:
+
+  struct CompareLevel {
+  public:
+    bool operator()(const TrigFTF_GNN_Edge* pS1, const TrigFTF_GNN_Edge* pS2) {
+      return pS1->m_level > pS2->m_level;
+    }
+  };
+
+ TrigFTF_GNN_Edge() : m_n1(nullptr), m_n2(nullptr), m_level(-1), m_next(-1), m_nNei(0) {};
+
+ TrigFTF_GNN_Edge(const TrigFTF_GNN_Edge<space_point_t>&  e) : m_n1(e.m_n1), m_n2(e.m_n2) {};
+
+  inline void initialize(TrigFTF_GNN_Node<space_point_t>* n1, TrigFTF_GNN_Node<space_point_t>* n2) {
+    m_n1 = n1; 
+    m_n2 = n2;
+    m_level = 1;
+    m_next = 1;
+    m_nNei = 0;
+  }
+
+
+  TrigFTF_GNN_Node<space_point_t>* m_n1{nullptr};
+  TrigFTF_GNN_Node<space_point_t>* m_n2{nullptr};
+  
+  signed char m_level{-1}, m_next{-1};
+
+  unsigned char m_nNei{0};
+  float m_p[4]{};
+  
+  unsigned int m_vNei[N_SEG_CONNS]{};//global indices of the connected edges
```

Comment:
will `m_p` and `m_vNei` be used later? They don't seem to be filled or used at the moment.

Reply:
Both used later in create seeds